### PR TITLE
Add/typescript

### PR DESCRIPTION
**Pull request by josipbrcina**
This PR converts codebase into `typescript`.  This is the first iteration, tests and storybook are excluded and will be addressed in next PRs

### CODE REVIEW

File: src/utils/getDefaultDateValue.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
-export const getDefaultDateValue = (currentDate) => {
+export const getDefaultDateValue = (currentDate?: Date): string => {
   const date = currentDate ?? new Date();
 
-  let day = date.getDate();
-  let month = date.getMonth() + 1;
+  let day: number | string = date.getDate();
+  let month: number | string = date.getMonth() + 1;
```
Comment (izebb) [inline src/utils/getDefaultDateValue.ts:5]: This should be `number`?   also we can also make use of ts inference and not specify any type

File: src/lib/Form.tsx [hunk 2]
```diff
@@ -1,28 +1,21 @@
...
-  children, bindUseForm, ...props
-}) => {
+import React, { ReactElement } from 'react';
+import { eventHandlers } from './enums';
+import { Obj, IForm, EventHandler } from './index';
+import { noop } from '../utils/noop';
+
+export const Form: React.FC<IForm> = ({
+  children = [], bindUseForm, ...props
+}): JSX.Element => {
```
Comment (izebb) [inline src/lib/Form.tsx:8]: We don't have to specify the type `JSX.Element`?

File: src/lib/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import React, {
+  ChangeEvent, MutableRefObject, ReactElement, ReactNode,
+} from 'react';
+import { htmlAttributes } from './enums';
+
+export type Obj = { [key: string]: Value };
```
Comment (izebb) [inline src/lib/index.d.ts:6]: another idea would be 
```tsx
  export type Obj = Record<string, Value>;
```

**Review by izebb — DISMISSED** [review]
Perfetto 🚀🚀

File: src/utils/getDefaultDateValue.ts (hunk 1 above)
Comment (josipbrcina) [reply src/utils/getDefaultDateValue.ts]: Few lines below this number is casted to a string via `template literals` and the function is returning a string. Refactoring -> will add parsedMonth and day in separate variables and leave this to be a number by ts inference! 👍

**Comment by josipbrcina** [comment]
> Perfetto 🚀🚀

Thx! 🥇

**Pull request merged**